System/GraphicalPrimitive.cpp

```diff
@@ -31,13 +31,43 @@ namespace RTE {
 		drawRightPos.m_Y -= targetPos.m_Y;
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	static void lineWithThickness(BITMAP *drawScreen, const Vector &startPos, const Vector &endPos, unsigned char color, const int &thickness) {
+		int x_delta = startPos.GetFloorIntX() - endPos.GetFloorIntX(),
+			y_delta = startPos.GetFloorIntY() - endPos.GetFloorIntY();
+		Vector n(-y_delta, x_delta);
+
+		int startIdx = 0;
```

Comment:
I think this odd number handling could be redone to avoid using abnormal for loops, and also make things a little more obvious:
```cpp
bool oddThickness = thickness % 2;
if (oddThickness) { line(drawScreen, startPos.GetFloorIntX(), startPos.GetFloorIntY(), endPos.GetFloorIntX(), endPos.GetFloorIntY(), color); }

for (int i = oddThickness ? 1 : 0; i < thickness / 2; ++i) {
	...
```